File: javascript/75.Sort-Colors.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var sortColors = function (nums) {
  let low = 0;
  let mid = 0;
  let high = nums.length - 1;

  while (mid <= high) {
    if (nums[mid] === 0) {
      [nums[low], nums[mid]] = [nums[mid], nums[low]];
      low += 1;
      mid += 1;
    } else if (nums[mid] === 1) {
      mid += 1;
    } else {
      [nums[mid], nums[high]] = [nums[high], nums[mid]];
      high -= 1;
    }
  }
};

File: python/75.Sort-Colors.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        n = len(nums)
        index = n - 1
        zindex = n - 1

        while index >= 0:
            if nums[index] != 0:
                nums[index], nums[zindex] = nums[zindex], nums[index]
                zindex -= 1
            index -= 1

        index = zindex + 1
        tindex = zindex + 1

        while index < n:
            if nums[index] != 2:
                nums[index], nums[tindex] = nums[tindex], nums[index]
                tindex += 1
            index += 1